includes/discovery/functions.inc.php

```diff
@@ -76,19 +76,14 @@ function discover_new_device($hostname, $device = [], $method = '', $interface =
     }
 
     try {
-        $remote_device_id = addHost($hostname, '', '161', 'udp', Config::get('default_poller_group'));
+        $remote_device_id = addHost($hostname, '', '161', 'udp', $device['poller_group']); // discover with actual poller group
         $remote_device = device_by_id_cache($remote_device_id, 1);
         echo '+[' . $remote_device['hostname'] . '(' . $remote_device['device_id'] . ')]';
         discover_device($remote_device);
         device_by_id_cache($remote_device_id, 1);
 
         if ($remote_device_id && is_array($device) && ! empty($method)) {
-            $extra_log = '';
-            $int = cleanPort($interface);
-            if (is_array($int)) {
-                $extra_log = ' (port ' . $int['label'] . ') ';
-            }
-
+            $extra_log = isset($interface) ? ' (port ' . htmlentities($interface) . ') ' : '';
```

Comment:
FYI, $interface is never a string.  EXCEPT when it is not set because of the braindead function defintion here. `$interface = ''`

htmlentities($interface) will fail because it is an array (when port is actually given to the function)  otherwise, it will just add ` (port )` to the log because it will always be an empty string.

Reply:
cleanPort() will set $interface['label'] which is not a database field.  (Which is why it is called)

Reply:
FYI, I found out the above by looking at all the calls to discover_new_device()